common/src/JetIds.cxx

```diff
@@ -53,54 +53,229 @@ bool DeepCSVBTag::operator()(const Jet & jet, const Event &) const{
+  if(jet.muonEnergyFraction() <0.8
+     &&jet.chargedEmEnergyFraction()<0.80)
+    return true;
+  return false;
+}
+
+//according to https://twiki.cern.ch/twiki/bin/viewauth/CMS/JetID13TeVRun2018
+bool JetPFID::tightID2018_CHS(const Jet & jet) const{
+  if(fabs(jet.eta())<=2.6
+     && jet.numberOfDaughters()>1 
+     && jet.neutralHadronEnergyFraction()<0.90
+     && jet.neutralEmEnergyFraction()<0.90
+     && jet.chargedHadronEnergyFraction()>0
+     && jet.chargedMultiplicity()>0){
+  }
+  else if(fabs(jet.eta())>2.6 && fabs(jet.eta())<=2.7
+	  && jet.neutralHadronEnergyFraction()<0.90 
+	  && jet.neutralEmEnergyFraction()<0.99
+	  && jet.chargedMultiplicity()>0){
+    return true;
+  }
+  else if(fabs(jet.eta())>2.7 && fabs(jet.eta())<=3
+	  &&jet.neutralEmEnergyFraction()<0.99
+	  &&jet.neutralEmEnergyFraction()>0.02
+	  &&jet.neutralMultiplicity()>2){
+    return true;
+  }
+  else if(fabs(jet.eta())>3
+	  && jet.neutralMultiplicity()>10
+	  && jet.neutralEmEnergyFraction()<0.90
+	  && jet.neutralHadronEnergyFraction()>0.02){
+    return true;
+  }
   return false;
 }
 
-bool JetPFID::tightLepVetoID(const Jet & jet) const{
-  if(!tightID(jet))return false;
+bool JetPFID::tightLepVetoID2018(const Jet & jet) const{
+  if(fabs(jet.eta())<=2.6 && !tightID2018_CHS(jet)) return false;
   if(jet.muonEnergyFraction() <0.8
      &&jet.chargedEmEnergyFraction()<0.80)
     return true;
   return false;
 }
+
+bool JetPFID::looseID2016_CHS(const Jet & jet) const{
+  if(fabs(jet.eta())<=2.7
+     && jet.numberOfDaughters()>1 
+     && jet.neutralHadronEnergyFraction()<0.99
+     && jet.neutralEmEnergyFraction()<0.99){
+    
+    if(fabs(jet.eta())>=2.4)
+      return true;
+      
+    if(jet.chargedEmEnergyFraction()<0.99
+       && jet.chargedHadronEnergyFraction()>0
+       && jet.chargedMultiplicity()>0)
+      return true;   
+  }
+  else if(fabs(jet.eta())>2.7 && fabs(jet.eta())<=3
+	  &&jet.neutralEmEnergyFraction()<0.90
+	  &&jet.neutralMultiplicity()>2){
+    return true;
+  }
+  else if(fabs(jet.eta())>3
+	  && jet.neutralMultiplicity()>10
+	  && jet.neutralEmEnergyFraction()<0.90){
+    return true;
+  }
+  return false;
+}
+
+bool JetPFID::tightID2016_CHS(const Jet & jet) const{
+  if(!looseID2016_CHS(jet)) return false;
+  if(fabs(jet.eta())<=2.7 
+     && jet.neutralEmEnergyFraction()<0.90
+     && jet.neutralHadronEnergyFraction()<0.90){
+    return true;
+  }
+  else if(fabs(jet.eta())>2.7){//the same as in looseID2016_CHS, thus checked
+    return true;
+  }
+  return false;
+}
+
+bool JetPFID::looseID2016_PUPPI(const Jet & jet) const{
+  if(fabs(jet.eta())>2.7) return true;
+  return looseID2016_CHS(jet);
+}
+
+bool JetPFID::tightID2016_PUPPI(const Jet & jet) const{
+  if(fabs(jet.eta())>2.7) return true;
+  return tightID2016_CHS(jet);
+}
+
+
+
+bool JetPFID::tightLepVetoID2016(const Jet & jet) const{
+  if(fabs(jet.eta())>2.7) return true;//no requirement for eta>2.7
```

Comment:
I never know whether they don't put any requirement on the twiki page on the assumption people just revert to the normal tight ID? i.e. would it be better to do here `if(fabs(jet.eta())>2.7) return tightID2016_CHS(jet);` ?

Reply:
This is the problem, not clear if default should be treated as CHS or as PUPPI